app/docs/md/learn/concepts/lifecycle.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Enhance Lifecycle
+---
+
+<doc-callout level="info" mark="📍">
+
+The journey of an HTTP request through Enhance isn't complicated, but it's helpful to understand how the final HTML document is assembled from the code you author in `app/`.
+
+</doc-callout>
+
+<doc-enhance-diagram class="block mb2">
+</doc-enhance-diagram>
+
+## Routing
+
+When an HTTP request (GET, POST, etc.) arrives, Enhance decides how to route it. First checking for [a matching API function](/docs/learn/starter-project/api) and then [a Page](/docs/learn/starter-project/pages).
+
+```
+my-rad-app.net/login --> app/api/login.mjs --> /app/pages/login.html
+```
+
+### Error Handling
+
+If an error is encountered, either because no route match was found (`404`) or an internal exception (`500`) occurs, Enhance calls [the corresponding error function](/docs/learn/starter-project/404-errors).
+
+## API Data Function
+
+When a requested route has a matching [`app/api/` file](/docs/learn/starter-project/api), it is passed the request and executed first so that [the Store](/docs/learn/concepts/state/store) can be populated for [the matching Page](/docs/learn/starter-project/pages).
+
+### Return JSON
+
+If there is an API function, but no matching Page or the request's headers includes `Accept: application/json`, the result of the API function will be returned directly to the client as JSON with the correct headers.
+
+<doc-callout level="tip" mark="🤖">
+
+This is a great way to add JSON API endpoints to your Enhance app! Even if those endpoints have an associated Page.
```

Comment:
Are you missing a `don't` in this sentence?

Reply:
No, you can get JSON from an API function even when there is a page. `Accept: application/json`

Reply:
Maybe it's just better to exclude this callout as to not muddy the waters.
The paragraphs above outline this feature of the lifecycle without making any conclusions about what it could be used for. It's essentially implied

Reply:
I updated the callout to be more clear:
> This is a great way to add JSON API endpoints to your Enhance app! Even if those endpoints have an associated Page: requests that set `Accept: application/json` will get plain JSON without a Page render.